guides/micronaut-data-jdbc-repository/micronaut-data-jdbc-repository.adoc

```diff
@@ -1,47 +1,63 @@
+* JDK 1.8 or greater installed with `JAVA_HOME` configured appropriately
+* https://www.docker.io/gettingstarted/#h_installation[Docker] installed to run MySQL and for running tests, using https://www.testcontainers.org[Testcontainers].
 
 include::{commondir}/common-completesolution.adoc[]
 
-include::{commondir}/common-create-app.adoc[]
+== Writing the Application
 
-=== Data Source configuration
+include::{commondir}/common-cli-or-launch.adoc[]
 
-Add the following dependencies:
+:exclude-for-languages:groovy
+[source,bash]
+----
+mn create-app --features=data-jdbc,flyway,jdbc-hikari,mysql,testcontainers,graalvm example.micronaut.micronautguide --build=@build@ --lang=@lang@
+----
+:exclude-for-languages:
 
-:dependencies:
+:exclude-for-languages:java,kotlin
+[source,bash]
+----
+mn create-app --features=data-jdbc,flyway,jdbc-hikari,mysql,testcontainers example.micronaut.micronautguide --build=@build@ --lang=@lang@
+----
+:exclude-for-languages:
 
-dependency:micronaut-data-processor[groupId=io.micronaut.data,scope=annotationProcessor,callout=1]
-dependency:micronaut-data-jdbc[groupId=io.micronaut.data,callout=2]
-dependency:micronaut-jdbc-hikari[groupId=io.micronaut.sql,callout=3]
-dependency:h2[groupId=com.h2database,scope=runtimeOnly,callout=4]
+include::{commondir}/common-build-lang-arguments.adoc[]
 
-:dependencies:
+:exclude-for-languages:groovy
+If you use Micronaut Launch, select Micronaut Application as application type and add the `data-jdbc`, `flyway`, `jdbc-hikari`, `mysql`, `testcontainers` and `graalvm` features.
+:exclude-for-languages:
+
+:exclude-for-languages:java,kotlin
+If you use Micronaut Launch, select Micronaut Application as application type and add the `data-jdbc`, `flyway`, `jdbc-hikari`, `mysql` and `testcontainers` features.
+:exclude-for-languages:
+
+include::{commondir}/common-default-package.adoc[]
 
-<1> Add Micronaut Data annotation processor
-<2> Configures a compile-scoped dependency on the `micronaut-data-jdbc`
-<3> Configures SQL DataSource instances using Hikari Connection Pool
-<4> Add dependency to in-memory H2 Database
+include::{commondir}/common-annotationprocessors.adoc[]
+
+=== Data Source configuration
+
+Add the following dependencies:
 
 Define the datasource in `src/main/resources/application.yml`.
 
 resource:application.yml[tag=datasource]
 
-NOTE: This way of defining the datasource properties enables us to externalize the configuration, for example for production environment, and also provide a default value for development. If the environment variables are not defined, the Micronaut framework will use the default values. +
-Also keep in mind that it is necessary to escape the `:` in the connection URL using backticks ```.
-
 === DB Schema
 
 We need a way to create the database schema. For that we will use
 https://micronaut-projects.github.io/micronaut-flyway/latest/guide/[Micronaut integration with Flyway].
 
-Add the following snippet to include the necessary dependencies:
```

Comment:
Instead of deleting this line, I would replace: 

> Add the following snippet to include the necessary dependencies:

with: 

> `flyway` feature adds the following dependencies: